src/spring/azext_spring/buildpack_binding.py

```diff
@@ -130,8 +132,10 @@ def _get_connection_string_from_app_insights(cmd, resource_group, app_insights):
     else:
         connection_string = _get_app_insights_connection_string(cmd.cli_ctx, resource_group, app_insights)
 
+    # Customer has specify the resourceId or application insights name.
+    # Raise exception when connection string not found in this scenario.
     if not connection_string:
-        logger.warning(
+        raise InvalidArgumentValueError(
```

Comment:
We could further enhance this case, move it to validator, so we can fail fast, rather than fail after service instance is created. We can enhance this in another PR.

Reply:
Will this cause breaking change after raising the `InvalidArgumentValueError`?